src/Bridge/Symfony/Bundle/DependencyInjection/Compiler/DataProviderPass.php

```diff
@@ -52,6 +53,11 @@ private function registerDataProviders(ContainerBuilder $container, string $type
         $queue = new \SplPriorityQueue();
 
         foreach ($services as $serviceId => $tags) {
+            $definition = $container->getDefinition($serviceId);
+            if (is_subclass_of($definition->getClass(), SerializerAwareDataProviderInterface::class)) {
+                $definition->addMethodCall('setSerializerLocator', [$container->getDefinition('api_platform.serializer_locator')]);
```

Comment:
I agree about the required dependencies in the constructor, but how can I properly set the locator in the constructor? It can be at any position or named with any variable name…
To do so, I did the same feature than the [Serializer](https://github.com/symfony/serializer/blob/master/Serializer.php#L65).